tests/pipeline/test_numerical_expression.py

```diff
@@ -147,6 +147,41 @@ def test_validate_bad(self):
         with self.assertRaises(TypeError):
             (f > f) > f
 
+    def test_many_inputs(self):
+        factors = []
+
+        # Create 11 factors to use as inputs. We need at least 11 to create a
+        # string expression containing an x_10. Then, ensure that this long
+        # expression can correctly merge with another.
+        for class_name in 'PQRSTUVWXYZ':
+            factors.append(
+                type(
+                    class_name,
+                    (Factor,),
+                    dict(dtype=float64_dtype, inputs=(), window_length=0),
+                )
+            )
+
+        short_num_expr = NumericalExpression(
+            "x_0", (self.f,), dtype=float64_dtype,
+        )
+        long_num_expr = NumericalExpression(
+            "x_0 + x_1 + x_2 + x_3 + x_4 + x_5 + x_6 + x_7 + x_8 + x_9 + x_10",
+            tuple(factors),
+            dtype=float64_dtype,
+        )
+
+        # Merge our long NumericalExpression with another.
+        long_expr, short_expr, new_inputs = \
+            long_num_expr.build_binary_op('+', short_num_expr)
+
+        # Test that it passes validation.
+        NumericalExpression(
+            ' + '.join([long_expr, short_expr]),
+            new_inputs,
+            dtype=float64_dtype,
+        )
```

Comment:
We should test actually computing these to make sure we don't scramble the variables.